assets/js/components/Dashboard/NewPaMessage/SelectStationsPage.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { useMemo, useState } from "react";
+import { Button, Container, Form } from "react-bootstrap";
+import fp from "lodash/fp";
+import { useScreenplayContext } from "Hooks/useScreenplayContext";
+import type { Place } from "Models/place";
+import RouteColumn from "./RouteColumn";
+import { GREEN_LINE_ROUTES, SILVER_LINE_ROUTES } from "Constants/constants";
+import StationGroupCheckbox from "./StationGroupCheckbox";
+import SelectedStationsSummary from "./SelectedStationsSummary";
+import {
+  GL_CENTRAL_SUBWAY,
+  GL_D_BRANCH,
+  GL_E_BRANCH,
+  GLX,
+  ORANGE_NORTH,
+  ORANGE_SOUTH,
+  RED_ASHMONT_BRANCH,
+  RED_BRAINTREE_BRANCH,
+  RED_TRUNK,
+} from "./StationGroups";
+import { Page } from "./types";
+
+const usePlacesWithPaEss = () => {
+  const { places } = useScreenplayContext();
+  return useMemo(
+    () =>
+      places
+        .map((place) => ({
+          ...place,
+          screens: place.screens.filter((screen) => screen.type === "pa_ess"),
+        }))
+        .filter((place: Place) => place.screens.length > 0),
+    [places],
+  );
+};
+
+const BASE_PLACE_ROUTE_TO_ROUTE_IDS: { [key: string]: string[] } = {
+  "Green-B": ["Green-B"],
+  "Green-C": ["Green-C"],
+  "Green-D": ["Green-D"],
+  "Green-E": ["Green-E"],
+  Red: ["Red"],
+  Orange: ["Orange"],
+  Blue: ["Blue"],
+  Mattapan: ["Mattapan"],
+  Silver: SILVER_LINE_ROUTES,
+};
+
+const ROUTE_TO_CLASS_NAMES_MAP: { [key: string]: string } = {
+  Red: "route-col--red",
+  Orange: "route-col--orange",
+  Blue: "route-col--blue",
+  Mattapan: "route-col--mattapan",
+  Silver: "route-col--silver",
+};
+
+interface Props {
+  navigateTo: (page: Page) => void;
+}
+
+const SelectStationsPage = ({ navigateTo }: Props) => {
+  const [zones, setZones] = useState<string[]>([]);
+  const places = usePlacesWithPaEss();
+  if (places.length === 0) return null;
+
+  const allRoutes = fp.uniq(
+    places.flatMap((place) =>
+      place.screens.flatMap((screen) => screen.route_ids ?? []),
+    ),
+  );
+
+  const busRoutes = fp.without(
+    Object.values(BASE_PLACE_ROUTE_TO_ROUTE_IDS).flat(),
+    allRoutes,
+  );
+
+  const PLACE_ROUTE_TO_ROUTE_IDS: { [key: string]: string[] } = {
```

Comment:
nit: capitalization a bit misleading here since this is not really a "constant" in the same sense as the others are, it's built anew each time this component is rendered, with potentially different results.

Reply:
Good call. I also feel like the name doesn't accurately represent the value so renamed it to `routeNameToRouteIds`.